recipe-client-addon/lib/NormandyDriver.jsm

```diff
@@ -156,6 +156,68 @@ this.NormandyDriver = function(sandboxManager) {
       sandboxManager.removeHold(`setTimeout-${token}`);
     },
 
+    /*
+     * Return a promise that resolves to an Addon ID if installation is successful.
+     */
+    installAddon(url) {
+        /* Install an addon.  Note that the AddonManager can only handle
+         * file:// and downloadable URLs.
+         */
+
+      var installObjectCallback = (installObject, resolve, reject) => {
+        installObject.install();
+        installObject.addListener({
+          onInstallEnded: (addonInstall, addon) => {
+            let addonId = addon.id;
+            resolve(addonId);
+          },
+          onInstallFailed: (addonInstall) => {
+            reject(`AddonInstall error code: [${addonInstall.error}]`);
+          },
+        });
+      };
+
+      function installObjectPromise(installObject) {
+        return new Promise(function(resolve, reject) {
+          installObjectCallback(installObject, resolve, reject);
+        });
+      }
+
+      function getInstallForURLPromise(url) {
+        return new Promise(function(resolve, reject) {
+          AddonManager.getInstallForURL(url, resolve, 'application/x-xpinstall');
+        });
+      };
+
+      // Return a promise
```

Comment:
I don't think so - I'd rather be consistent with the use of promises instead of a mix of async functions and promise objects so that callers know what to expect.